core/src/Cuda/Kokkos_Cuda_Instance.hpp

```diff
@@ -373,26 +371,40 @@ namespace Experimental {
 //   Customization point for backends
 //   Default behavior is to return the passed in instance
 
+namespace Impl {
+// Create new Cuda space on the device of base_space
+inline Cuda create_cuda_space(const Cuda& base_space) {
+  cudaStream_t stream;
+  KOKKOS_IMPL_CUDA_SAFE_CALL(
+      (base_space.impl_internal_space_instance()->cuda_stream_create_wrapper(
+          &stream)));
+  return Cuda(stream, Kokkos::Impl::ManageStream::yes);
+}
+}  // namespace Impl
+
 template <class... Args>
-std::vector<Cuda> partition_space(const Cuda&, Args...) {
+std::array<Cuda, sizeof...(Args)> partition_space(const Cuda& cuda_space,
+                                                  Args... ignored) {
   static_assert(
       (... && std::is_arithmetic_v<Args>),
       "Kokkos Error: partitioning arguments must be integers or floats");
-  std::vector<Cuda> instances(sizeof...(Args));
-  Kokkos::Impl::create_Cuda_instances(instances);
-  return instances;
+  return {((ignored, Impl::create_cuda_space(cuda_space)), ...)};
```

Comment:
I would prefer readability over perfromance considerations or brevity here and use create an `std::array` with default instances first and the assign new ones (like you had initially). After all the creation of a new execution space instances should be much more expensive then creating a copy of the default execution space instance.

Reply:
I would have suppressed the warning.  I disagree but won't block.